chia/data_layer/data_layer.py

```diff
@@ -241,8 +241,12 @@ async def fetch_and_validate(self, subscription: Subscription) -> None:
             f"Target wallet generation: {singleton_record.generation}."
         )
 
-        downloaded = await download_data(self.data_store, subscription, singleton_record.root)
+        try:
+            downloaded = await download_data(self.data_store, subscription, singleton_record.root)
+        except Exception:
```

Comment:
Should probably list the Exceptions here, eg. `RuntimeError`, `HttpWhatever`, ...

Reply:
Yeah, this will consume cancellation on Python 3.7.  It also totally silently hides typos etc.  This is a very general thing we need to discuss across the code base.